app/public/js/localSprites.js

```diff
@@ -141,6 +144,35 @@ function updateLocalSprite(localSprite) {
         // This flag will be used in the update loop to remove this sprite from the list of localSprites.
         localSprite.shouldBeRemoved = true;
     }
+    //geomtry collision checks for projectiles that die on impact
+    //BROKEN: It looks to me like "alignToTile" isn't happening, and the fireball is detonating in the frame before it would impact something,
+    //  rather than right next to the wall it would be moving past.
+    if (localSprite.vx && localSprite.diesOnImpact) {
+        if (localSprite.vx < 0) {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vx, TILE_LEFT))) localSprite.shouldBeRemoved = true;
+        } else {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vx, TILE_RIGHT))) localSprite.shouldBeRemoved = true;
+        }
+    }
+    if (localSprite.vy && localSprite.diesOnImpact) {
+        if (localSprite.vy < 0) {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vy, TILE_UP))) localSprite.shouldBeRemoved = true;
+        } else {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vy, TILE_DOWN))) localSprite.shouldBeRemoved = true;
+        }
+    }
+    //update trigger zones
+    if (localSprite.name === 'triggerZone') {
+        if (rectanglesOverlap(localSprite.hitBox, getGlobalSpriteHitBox(localSprite.target)) && localSprite.cooldownTimer === 0) { //WRONG, probably: sould probabaly use getGlobalSpriteHitBox(localSprite), but not sure how that will interact with triggerZones lack of meaningful frames data.
```

Comment:
Since it apparently has `hitBox` directly on the sprite, it won't fall back to trying to find it on the frames, so you can just use `getGlobalSpriteHitBox` here. Then it will work whether it has it on the sprite or has frames.